gateway/src/inference/providers/fireworks.rs

```diff
@@ -48,24 +48,12 @@ impl FireworksProvider {
         model_name: String,
         api_key_location: Option<CredentialLocation>,
     ) -> Result<Self, Error> {
-        let api_key_location = api_key_location.unwrap_or(default_api_key_location());
-        let credentials = match api_key_location {
-            CredentialLocation::Env(key_name) => {
-                let api_key = env::var(key_name)
-                    .map_err(|_| ErrorDetails::ApiKeyMissing {
-                        provider_name: "Fireworks".to_string(),
-                    })?
-                    .into();
-                FireworksCredentials::Static(api_key)
-            }
-            CredentialLocation::Dynamic(key_name) => FireworksCredentials::Dynamic(key_name),
-            _ => Err(Error::new(ErrorDetails::Config {
-                message: "Invalid api_key_location for Fireworks provider".to_string(),
-            }))?,
-        };
+        let credential_location = api_key_location.unwrap_or(default_api_key_location());
+        let generic_credentials = Credential::try_from((credential_location, "Fireworks"))?;
```

Comment:
We are mixing types and names (e.g. `anthropic`, `Fireworks`). Let's use types throughout, i.e. should match the `type` in config.